src/Atc/Extensions/BaseTypes/DateTimeOffsetExtensions.cs

```diff
@@ -11,16 +11,52 @@ public static class DateTimeOffsetExtensions
         /// <summary>
         /// Returns true if the date time offset is between or equal to one of the two values.
         /// </summary>
-        /// <param name="date">DateTime Base, from where the calculation will be preformed.</param>
+        /// <param name="dateTimeOffset">DateTime Base, from where the calculation will be preformed.</param>
         /// <param name="startDate">Start date to check for.</param>
         /// <param name="endDate">End date to check for.</param>
         /// <returns>boolean value indicating if the date is between or equal to one of the two values.</returns>
-        public static bool IsBetween(this DateTimeOffset date, DateTimeOffset startDate, DateTimeOffset endDate)
+        public static bool IsBetween(this DateTimeOffset dateTimeOffset, DateTimeOffset startDate, DateTimeOffset endDate)
         {
-            var ticks = date.Ticks;
+            var ticks = dateTimeOffset.Ticks;
             return ticks >= startDate.Ticks && ticks <= endDate.Ticks;
         }
 
+        /// <summary>
+        /// Resets to start of current hour.
+        /// </summary>
+        /// <param name="dateTimeOffset">The date time offset.</param>
+        /// <returns>The dateTimeOffset with the seconds and milliseconds as 0.</returns>
+        public static DateTimeOffset ResetToStartOfCurrentHour(this DateTimeOffset dateTimeOffset)
```

Comment:
Consider adding null check.

Reply:
It is a extension method to DateTimeOffset and not DateTimeOffset?
If I try to make a null check - I get error from rule CS8073 and S2583 - So null check is not allowed.